data.tf

```diff
@@ -1,18 +1,23 @@
 data "aws_region" "current" {}
 data "aws_caller_identity" "current" {}
 
-# Most recent Amazon Linux 2 AMI
-data "aws_ami" "amazon_linux_2" {
+# Most recent Amazon Linux 2023 AMI
+data "aws_ami" "amazon_linux_2023" {
   most_recent = true
   owners      = ["amazon"]
 
   filter {
     name   = "name"
-    values = ["amzn2-ami-hvm*"]
+    values = ["al2023-ami*"]
```

Comment:
Their naming change of 2023 is interesting, before they just named it as AL2, now they're naming as AL2023